plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -319,18 +322,33 @@ protected String renameTableSql(String catalogName, String remoteSchemaName, Str
     }
 
     @Override
-    protected String renameColumnSql(JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newRemoteColumnName)
+    public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newColumnName)
     {
         RemoteTableName remoteTableName = handle.asPlainTable().getRemoteTableName();
-        return format(
-                "sp_rename %s, %s, 'COLUMN'",
-                singleQuote(remoteTableName.getCatalogName().orElse(null), remoteTableName.getSchemaName().orElse(null), remoteTableName.getTableName(), "[" + escape(jdbcColumn.getColumnName()) + "]"),
-                "[" + newRemoteColumnName + "]");
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            String columnFrom = DOT_JOINER.join(remoteTableName.getCatalogName().orElse(null), remoteTableName.getSchemaName().orElse(null), quoteEscapeForSQLServer(remoteTableName.getTableName()), quoteEscapeForSQLServer(jdbcColumn.getColumnName()));
```

Comment:
according to my very fast test `join` throws NullPointerException on null argument. Please doublecheck that.

Reply:
It's kind of previous functionality, which we had before in:
`private static String singleQuote(String... objects)
    {
        return singleQuote(DOT_JOINER.join(objects));
    }`

Not sure what we want here

Reply:
The question is if this is a legal state here that remoteTableName does not have catalog or schema. 
If yes - join has to be performed differently.
If no - it worth to throw as soon as possible.

Reply:
I checked where is values come, looks like from `RemoteTableName`

```
 public RemoteTableName(Optional<String> catalogName, Optional<String> schemaName, String tableName)
    {
        this.catalogName = requireNonNull(catalogName, "catalogName is null");
        this.schemaName = requireNonNull(schemaName, "schemaName is null");
        this.tableName = requireNonNull(tableName, "tableName is null");
    }
```
So it's not possible to have null in this fields, so looks like we are safe

Reply:
regarding your code quote: it's not possible to have null values, but it possible to have empty ones.
regarding `.join` method - it obtains optionals which are tested to have value (yes, they are not null), and if they are empty - return null. So join fails.

Reply:
I tried to create schema in SQL Server with empty name, looks like it's not possible.
So I thunk we are ok to return null and then through exception in result is empty.

Reply:
> Maybe we can replace
remoteTableName.getCatalogName().orElse(null)
with
remoteTableName.getCatalogName().orElseThrow()
?

👍